src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,122 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    public static final double RESIZE_FACTOR = 1.5;
+
+    private T[] container;
+    private int size;
+
+    public ArrayList() {
+        container = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growArray(size + 1);
```

Comment:
You call method only with size+1 agrument. Better just include this value to method logic
```suggestion
        growIfRequired();
```